paper.tex

```diff
@@ -35,6 +35,8 @@
 
 \usepackage{longtable}
 
+\hyphenation{Math-e-ma-ti-ca pol-y-morph-ism}
```

Comment:
That's not the hyphenation that I got here: https://www.hyphenation24.com/?term=Mathematica, even though it said it can't find the word, so it might be inaccurate.

As to `polymorphism`, again, Webster http://www.merriam-webster.com/dictionary/polymorphism suggests poly·mor·phism, which is different to what you put there.